src/main/kotlin/com/fasterxml/jackson/module/kotlin/ArgumentBucket.kt

```diff
@@ -0,0 +1,91 @@
+package com.fasterxml.jackson.module.kotlin
+
+import kotlin.reflect.KParameter
+
+internal class BucketGenerator(parameters: List<KParameter>) {
+    private val paramSize: Int = parameters.size
+    val maskSize = (paramSize / Int.SIZE_BITS) + 1
+    // For Optional and Primitive types, set the initial value because the function cannot be called if the argument is null.
+    private val originalValues: Array<Any?> = Array(paramSize) {
+        val param = parameters[it]
+
+        if (param.isOptional) {
+            ABSENT_VALUE[param.type.erasedType()]
+        } else {
+            null
+        }
+    }
+    private val originalMasks: IntArray = IntArray(maskSize) { FILLED_MASK }
+
+    fun generate() = ArgumentBucket(paramSize, originalValues.clone(), originalMasks.clone())
+
+    companion object {
+        private const val FILLED_MASK = -1
+
+        private val ABSENT_VALUE: Map<Class<*>, Any> = mapOf(
+            Byte::class.javaPrimitiveType!! to Byte.MIN_VALUE,
```

Comment:
If this is to contain all wrapper values, need to also add `Character` and `Boolean`?

Reply:
Yes, I forgot about them.
Fixed.
bc3581a